zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/TestRawZkClient.java

```diff
@@ -558,6 +558,41 @@ public void testCreateEphemeralWithValidSession()
     _zkClient.delete(path);
   }
 
+  /*
+   * This test validates that when ZK_AUTOSYNC_ENABLED_DEFAULT is enabled, sync() would be issued
+   * before handleNewSession. ZKclient would not see stale data.
+   */
+  @Test
+  public void testAutoSyncWithNewSessionEstablishment() throws Exception {
```

Comment:
This test seems is not quite relevant, how can we at least make sure the sync call success from this test?

Reply:
This test is just to make sure the normal flow will work. Namely no hanging. This is to make sure the change will not fail in the field after session expiration. This is a "safety" test.

The other side is to make sure not seeing outdate data if a follower Zk lagging leader. This can't be easily done without touch code of ZK server. Here, we have to trust sync() implementation is right.